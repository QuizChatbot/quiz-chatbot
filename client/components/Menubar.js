import React, { PropTypes, Component } from 'react'
import { Link } from 'react-router-dom'

export function renderLoginButton(onLoginClick) {
  return (<button onClick={() => onLoginClick()}>Login with Facebook</button>)
}

export function renderLogoutButton(onLogoutClick) {
  return (<button onClick={() => onLogoutClick()}>Logout</button>)
}

export class Menubar extends Component {
  constructor(props, context) {
    super(props, context)
  }

  render() {
    const { firedux, authed, onLoginClick, onLogoutClick } = this.props
    return (<div>
      <Link to="/">Leaderboard</Link><br />
      {!authed
        ? (renderLoginButton(onLoginClick))
        : (<div>
          <span> name: {firedux.displayName} </span><br />
          <Link to="/myquiz">My Quiz</Link><br />
          {renderLogoutButton(onLogoutClick)}
        </div>)
      }
    </div>
    )
  }
}

Menubar.PropTypes = {
  firedux: PropTypes.object.isRequired,
  authed: PropTypes.bool.isRequired,
  onLoginClick: PropTypes.func,
  onLogoutClick: PropTypes.func
}

export default Menubar