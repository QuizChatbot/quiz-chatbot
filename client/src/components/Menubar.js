import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import FlatButton from 'material-ui/FlatButton'
import RaisedButton from 'material-ui/RaisedButton'
import Paper from 'material-ui/Paper'

const style = {
  textAlign: 'center',
  display: 'inline-block'
}

export const LoginButton = ({ onLoginClick }) => (
  <RaisedButton label='Login with Facebook' primary onTouchTap={onLoginClick} />
)

export const LogoutButton = ({ onLogoutClick }) => (
  <RaisedButton label='Logout' primary onTouchTap={onLogoutClick} />
)

export class Menubar extends Component {
  render () {
    const { firedux, authed, onLoginClick, onLogoutClick } = this.props
    return (
      <div>
        <Paper style={style} zDepth={1} rounded={false}>

          <FlatButton label='LEADERBOARD' containerElement={<Link to='/' />} />
          <FlatButton
            style={{ verticalAlign: 'top' }}
            label='PLAY QUIZ'
            href='https://www.facebook.com/messages/t/122419575009686'
            target='_blank'
          />
          <FlatButton
            label='MY QUIZ'
            disabled={!authed}
            containerElement={<Link to='/myquiz' />}
          />
        </Paper>
        {!authed
          ? <LoginButton onLoginClick={onLoginClick} />
          : <div>
            <span>name: {firedux.displayName}</span>
            <LogoutButton onLogoutClick={onLogoutClick} />
          </div>}
      </div>
    )
  }
}

Menubar.PropTypes = {
  firedux: PropTypes.object,
  authed: PropTypes.bool,
  onLoginClick: PropTypes.func,
  onLogoutClick: PropTypes.func
}

export default Menubar
