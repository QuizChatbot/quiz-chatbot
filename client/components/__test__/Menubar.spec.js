import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { Menubar, renderLoginButton, renderLogoutButton } from '../Menubar'
import { Link } from 'react-router-dom'


describe('Menubar component test', () => {

  it('should always render leaderboard link', () => {
    const props = {
      firedux: {
        displayName: "Xiao Jia Ying"
      },
      auth: true
    }
    const Wrapper = shallow(<Menubar {...props} />)
    expect(Wrapper.contains(<Link to="/">Leaderboard</Link>)).toMatchSnapshot()
  })

  it('should render login button if authed is false', () => {
    const props = {
      auth: false
    }
    const Wrapper = shallow(<Menubar {...props} />)
    expect(Wrapper.contains('Login with Facebook')).toMatchSnapshot()
    expect(Wrapper.contains('Logout')).toMatchSnapshot()
  })

  it('should render user name and logout button if authed is true', () => {
    const props = {
      firedux: {
        displayName: "Xiao Jia Ying"
      },
      auth: true
    }
    const Wrapper = shallow(<Menubar {...props} />)
    expect(Wrapper.contains('Login with Facebook')).toMatchSnapshot()
    expect(Wrapper.contains('name: Xiao Jia Ying')).toMatchSnapshot()
    expect(Wrapper.contains('Logout')).toMatchSnapshot()
  })

  it('should render Login button if renderLoginButton function is called', () => {
    const func = () => { return }
    const Wrapper = shallow(<renderLoginButton onLoginClick={func} />)
    expect(Wrapper.contains(<button onClick={() => onLoginClick()}>Login with Facebook</button>)).toMatchSnapshot()
  })

  it('should render Logout button if renderLogoutButton function is called', () => {
    const func = () => { return }
    const Wrapper = shallow(<renderLogoutButton onLogoutClick={func} />)
    expect(Wrapper.contains(<button onClick={() => onLogoutClick()}>Logout</button>)).toMatchSnapshot()
  })

})