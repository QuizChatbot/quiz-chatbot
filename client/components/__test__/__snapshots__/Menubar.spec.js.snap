// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menubar component test should always render leaderboard link 1`] = `true`;

exports[`Menubar component test should render Login button if renderLoginButton function is called 1`] = `false`;

exports[`Menubar component test should render Logout button if renderLogoutButton function is called 1`] = `false`;

exports[`Menubar component test should render login button if authed is false 1`] = `true`;

exports[`Menubar component test should render login button if authed is false 2`] = `false`;

exports[`Menubar component test should render user name and logout button if authed is true 1`] = `true`;

exports[`Menubar component test should render user name and logout button if authed is true 2`] = `false`;

exports[`Menubar component test should render user name and logout button if authed is true 3`] = `false`;
